www/source/blog/2017-11-16-angular-seed.html.md

```diff
@@ -0,0 +1,117 @@
+---
+title: NodeJS Scaffolding with Angular Seed!
+date: 2017-11-16
+author: Nell Shamrell-Harrington
+tags: Scaffolding, Node, Angular
+category: community
+classes: body-article
+---
+
+Hello my fellow Habicats!
+
+Do you have legacy Angular projects that use [angular-seed](https://github.com/mgechev/angular-seed)? Now you can package them with Habitat using the Node Scaffolding! Let's give it a go!
```

Comment:
I'm not sure "legacy projects that use Angular Seed" makes sense, since Angular Seed is a starter kit. It might be better to consider aiming this at developers of Angular apps rather than strictly those using Angular Seed specifically (many may not -- I actually never have, though I know it's widely referenced), and just use Angular seed as the reference example?

Reply:
That was something I added later on in the process - I've noticed that the moment I mention Angular Seed, someone tends to bring up that Angular does not recommend its use anymore in favor of angular cli.  That said, Habitat is intended to be used with legacy applications, and we need a way to support them.

In short, I was trying to preempt the "Well, actually, you shouldn't be using Angular Seed anyway..." comments.  I'll consider a better way to word this.

Reply:
Yeah that's all true, which is why it might be good to tailor this more to a general Angular audience than specifically to Angular Seed or angular-cli, since scaffolding *should* be able to support either one.

I'm starting to wonder whether we can get away without having some sort of additional configuration setting in which the scaffolding user specifies which `npm` command should be run for `build` and `start` steps. I don't know there's going to be a reliable way to detect whether, for example, `build` should be run in all cases (because here, it's `build.prod`) or whether it makes sense to use `start` as a trigger to create a Habitat service (as here, it isn't, but for a Node app running Express or Koa, it would be).  Hrm.

Reply:
I'm planning on doing a follow up that highlights angular-cli - I'll mention that at the bottom of this blog post.  Do you mind posting your thoughts on additional configuration setup as an issue in core-plans?  That way we can track it and continue the conversation :)

While I would LOVE to do a general blog post that covers all things Angular, I think there is too much divergence in how it is used (starter kits vs. yeoman vs. angular-cli) to do a one size fits most (definitely not all!) post at this time.  I'd like to have this be the start of a series on packaging Angular apps using the node scaffolding and cover each common way individually (especially since I will likely need to make changes to the scaffolding to accommodate each of them, I haven't tested that yet).